### PR TITLE
Correctly handle empty password in AcquireUserCredentials

### PR DESCRIPTION
**Pull request by enj**
The current code will panic with an index out of bound error when an empty password is passed to AcquireUserCredentials.  It assumes that p is always a valid slice but it is nil when no password is given.  The panic occurs at `Password: &p[0]`.

This change removes the special handling of the empty password.  Thus `syscall.UTF16FromString(password)` is always called which guarantees that p is always a valid slice.

Signed-off-by: Monis Khan <mkhan@redhat.com>

### CODE REVIEW

File: negotiate/negotiate.go [hunk 1]
```diff
@@ -67,22 +67,17 @@ func AcquireUserCredentials(domain, username, password string) (*sspi.Credential
...
+	p, err := syscall.UTF16FromString(password)
+	if err != nil {
+		return nil, err
 	}
 	ai := sspi.SEC_WINNT_AUTH_IDENTITY{
 		User:           &u[0],
 		UserLength:     uint32(len(u) - 1), // do not count terminating 0
 		Domain:         &d[0],
 		DomainLength:   uint32(len(d) - 1), // do not count terminating 0
 		Password:       &p[0],
```
[the remark this answers is not in the document]
Comment (alexbrainman) [reply negotiate/negotiate.go]: Looking at https://msdn.microsoft.com/en-us/library/windows/desktop/aa378664(v=vs.85).aspx I think what you have done is fine. But would you mind also adding new test that verifies that function works with empty password? And add the same test in ntlm package.

[the remark this answers is not in the document]
Comment (enj) [reply negotiate/negotiate.go]: @alexbrainman it is a bit unclear to me where / how I would add such a test.  Do you want me to simply remove these lines?

https://github.com/alexbrainman/sspi/blob/4729b3d4d8581b2db83864d1018926e4154f9406/negotiate/negotiate_test.go#L210-L212

https://github.com/alexbrainman/sspi/blob/ec10aa0cc6e70a7b484df13af953a4d0b376f7f8/ntlm/ntlm_test.go#L109-L111
